compiler/next/include/chpl/queries/stringify-functions.h

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+  This file implements the generic chpl::stringify
+  as well as specializations for some common types.
+
+  stringify can be used to get a string representation of an object
+  for debugging purposes
+ */
+
+#ifndef CHPL_QUERIES_STRINGIFY_FUNCTIONS_H
+#define CHPL_QUERIES_STRINGIFY_FUNCTIONS_H
+
+#include <cassert>
+#include <cstring>
+#include <functional>
+#include <string>
+#include <unordered_map>
+#include <vector>
+#include "chpl/util/memory.h"
+#include <set>
+#include <sstream>
+
+
+namespace chpl {
+class Context;
+
+enum StringifyKind {
+  DEBUG_SUMMARY,
+  DEBUG_DETAIL,
+  CHPL_SYNTAX
+};
+
+
+template<typename T> struct stringify {
+  std::string operator()(StringifyKind stringKind, const T& stringMe) const = 0;
+};
+
+
+template<typename T> struct stringify<const T*> {
+  std::string operator()(StringifyKind stringKind, const T* stringMe) const {
+    stringify<T> stringifier;
+    return stringifier(stringKind, *stringMe);
+  }
+};
+
+template<typename T>
+static inline std::string defaultStringify(StringifyKind stringKind,
+                                           const T& stringMe) {
+
+  return "HAVE NOT IMPLEMENTED STRINGIFY YET";
```

Comment:
would be cool to get a stringname of the type T so it would be "have not implemented for ResolutionResult" or somthing

Reply:
I thought this would be nice too, but I wasn't sure of a reliable way to get the type name in a string. I didn't look too hard though, maybe this is simple?

Reply:
Haha if only things were simple :) I think type_info::name is only runtime and probably not worth bringing that in for this. I found a hacky c++17 way to do it [here](https://bitwizeshift.github.io/posts/2021/03/09/getting-an-unmangled-type-name-at-compile-time/) (boost licensed) but I'd say don't worry about it at this point

Reply:
I updated the callsites as suggested and removed `defaultStringify`